core/src/main/resources/error/README.md

```diff
@@ -0,0 +1,79 @@
+# Guidelines
+
+To throw a standardized exception, developers should use an error class and message parameters
```

Comment:
Shall we highlight "user-facing exception"? I think for assert-like errors we can still keep them unchanged in the Spark codebase.